client/signup/steps/courses/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { useViewportMatch } from '@wordpress/compose';
+import { useTranslate } from 'i18n-calypso';
+import React, { useState, useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import VideosUi from 'calypso/components/videos-ui';
+import StepWrapper from 'calypso/signup/step-wrapper';
+import { saveSignupStep, submitSignupStep } from 'calypso/state/signup/progress/actions';
+import CoursesFooter from './footer';
+import CoursesHeader from './header';
+import './style.scss';
+
+interface Props {
+	stepName: string;
+	goToNextStep: () => void;
+}
+
+export default function CoursesStep( props: Props ): React.ReactNode {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const { stepName, goToNextStep } = props;
+	const isMobile = useViewportMatch( 'small', '<' );
+	const [ isCourseComplete, setIsCourseComplete ] = useState( false );
+	const hideSkip = isMobile && isCourseComplete;
+
+	const onCourseComplete = useCallback( () => setIsCourseComplete( true ), [
+		setIsCourseComplete,
+	] );
+
+	const onStartWriting = () => {
+		dispatch( submitSignupStep( { stepName } ) );
+		goToNextStep();
+	};
+
+	React.useEffect( () => {
+		dispatch( saveSignupStep( { stepName } ) );
+	}, [] ); // eslint-disable-line react-hooks/exhaustive-deps
+
+	return (
+		<StepWrapper
+			className="courses"
+			isFullLayout
+			hideFormattedHeader
+			stepContent={
+				<VideosUi
+					headerBar={ <CoursesHeader /> }
+					footerBar={
+						<CoursesFooter
+							onCourseComplete={ onCourseComplete }
+							onStartWriting={ onStartWriting }
+						/>
```

Comment:
Oh I see now! It's because `<VideosUi>` is passing an `isCourseComplete` prop to the `footerBar` component. I think we should add an `onCourseComplete` callback to `<VideosUi>` to simplify the data flow. I think it's a reasonable sort of callback for this component to support? @andres-blanco @blackjackkent

Reply:
Hmmm...I create a hook named `useCourseData` to share the data such as `isCourseComplete`.